src/main/java/com/faforever/client/mod/ModService.java

```diff
@@ -307,45 +308,63 @@ public void overrideActivatedMods(Collection<ModVersionBean> modVersions) {
   private Set<String> readActiveMods() throws IOException {
     Path preferencesFile = preferencesService.getPreferences().getForgedAlliance().getPreferencesFile();
     Set<String> activeMods = new HashSet<>();
-
-    String preferencesContent = Files.readString(preferencesFile, StandardCharsets.UTF_8);
-    Matcher matcher = ACTIVE_MODS_PATTERN.matcher(preferencesContent);
-    if (matcher.find()) {
-      Matcher activeModMatcher = ACTIVE_MOD_PATTERN.matcher(matcher.group(0));
-      while (activeModMatcher.find()) {
-        String modUid = activeModMatcher.group(1);
-
-        if (Boolean.parseBoolean(activeModMatcher.group(2))) {
-          activeMods.add(modUid);
+    List<Charset> preferredEncodings = new ArrayList<>(List.of());
+    Map<String, Charset> map = Charset.availableCharsets();
+    for (Map.Entry<String, Charset> entry : map.entrySet()) {
+      Charset charset = entry.getValue();
+      preferredEncodings.add(charset);
+    }
+    for (Charset charset : preferredEncodings) {
+      try {
+        String preferencesContent = Files.readString(preferencesFile, charset);
+        Matcher matcher = ACTIVE_MODS_PATTERN.matcher(preferencesContent);
+        if (matcher.find()) {
+          Matcher activeModMatcher = ACTIVE_MOD_PATTERN.matcher(matcher.group(0));
+          while (activeModMatcher.find()) {
+            String modUid = activeModMatcher.group(1);
+            if (Boolean.parseBoolean(activeModMatcher.group(2))) {
+              activeMods.add(modUid);
+            }
+          }
         }
+        break;
+      } catch (MalformedInputException e) {
+        throw new AssetLoadException("Could not read mod state", e, "mod.errorReadingMods");
       }
     }
-
     return activeMods;
   }
 
   private void writeActiveMods(Set<String> activeMods) {
-    try {
-      Path preferencesFile = preferencesService.getPreferences().getForgedAlliance().getPreferencesFile();
-      String preferencesContent = Files.readString(preferencesFile, StandardCharsets.UTF_8);
+    List<Charset> preferredEncodings = new ArrayList<>(List.of());
+    Path preferencesFile = preferencesService.getPreferences().getForgedAlliance().getPreferencesFile();
+    Map<String, Charset> map = Charset.availableCharsets();
+    for (Map.Entry<String, Charset> entry : map.entrySet()) {
+      Charset charset = entry.getValue();
+      preferredEncodings.add(charset);
+    }
+    for (Charset charset : preferredEncodings) {
+      try {
+        String preferencesContent = Files.readString(preferencesFile, charset);
+        String currentActiveModsContent = null;
+        Matcher matcher = ACTIVE_MODS_PATTERN.matcher(preferencesContent);
+        if (matcher.find()) {
+          currentActiveModsContent = matcher.group(0);
+        }
```

Comment:
same as above. In fact might be best to make the reading of the pref file a function